lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -171,7 +172,14 @@ namespace Js
         {
             if (*exceptionVar == nullptr)
             {
-                *exceptionVar = JavascriptError::CreateFromCompileScriptException(scriptContext, &se);
+                if(this->GetModuleURL() != nullptr)
+                {
+                    *exceptionVar = JavascriptError::CreateFromCompileScriptException(scriptContext, &se, this->GetModuleURLSz());
+                }
+                else
+                {
+                    *exceptionVar = JavascriptError::CreateFromCompileScriptException(scriptContext, &se);                    
```

Comment:
I would prefer if this and the block below fetched the source URL before calling `JavascriptError::CreateFromCompileScriptException` so that we could make this a bit simpler. More like:

```c++
const WCHAR * sourceURL = nullptr;
if (this->GetModuleURL())
{
  sourceURL = this->GetModuleURLSz();
}
*exceptionVar = JavascriptError::CreateFromCompileScriptException(scriptContext, &se, sourceURL);
```